src/main/java/com/hutchison/vibe/model/BotState.java

```diff
@@ -67,6 +70,16 @@ public void play(CommandMessage commandMessage, MessageReceivedEvent event) {
         }
     }
 
+    public void togglePause(CommandMessage commandMessage, MessageReceivedEvent event) {
```

Comment:
Oh snap I get to leave a comment! :D

Anyways I'm not sure I like the `togglePause` idea. It's kind of impure, as each call oscillates the state. 

I think I would prefer a pause method that pauses if the track is playing and does nothing if the track is always paused, and likewise for resume. Something like:
`
    public void pause(CommandMessage commandMessage, MessageReceivedEvent event) {
        TrackScheduler scheduler = vibeAudioManager.getTrackScheduler();
        if (!scheduler.hasCurrentTrack()) return;
        if (scheduler.isPaused()) return;
        scheduler.togglePause();
    }
`
 
What do you think?

Reply:
By the way, I do see that's how the lavaplayer does it, and I still don't like it, lol